### ISSUE TITLE
`redefined-while-unused` (`F811`) - false negative on variable redeclaration

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#23802]
### Summary

```py
def foo(): ...
def foo(): ... # error

bar: int = 1
bar: int = 2 # no error
```

https://play.ruff.rs/6b9cbf95-6f20-49b8-bc54-42230fc95411

(specifically it's the type annotation that makes it a redeclaration)

### Version

0.15.5

**Comment by ntBre** [comment]
This sounds like a reasonable change to me. We could give it a try in preview.
